test/test_model.py

```diff
@@ -3791,5 +3791,82 @@ def test_repr(self):
         )
 
 
+class TestCloudCredential(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cloud_credential_dict = {
+            'auth-type': 'certificate',
+            'attrs': {
+                'client-cert': 'foo',
+                'client-key': 'bar',
+                'server-cert': 'baz'
+            },
+        }
+
+    def test_from_dict(self):
+        cloud_cred = ops.CloudCredential.from_dict(self.cloud_credential_dict)
+        self.assertEqual(cloud_cred.auth_type, 'certificate')
+        self.assertEqual(cloud_cred.attributes, self.cloud_credential_dict.get('attrs'))
+
+
+class TestCloudSpec(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cloud_credential_dict = {
```

Comment:
Similar here, let's just make this local to the test, and again have two tests, one with minimal required fields (optional fields not present rather than None), and one with all fields set.